tests/jballerina-unit-test/src/test/resources/test-src/types/regexp/regexp_value_test.bal

```diff
@@ -365,6 +365,44 @@ function testComplexRegExpValue2() {
 
     string:RegExp x5 = re `a\td-*ab[^c-f]+(?m:xj(?i:x|y))`;
     assertEquality("a\\td-*ab[^c-f]+(?m:xj(?i:x|y))", x5.toString());
+
+    string a = "A";
+    int b = 20;
+    string:RegExp x6 = re `abc${a}(?i:[^a-bcd-e]{1,}a{1,2}b{1}(?i:cd*${b*20}+)${b}+)${a+"C"}*`;
+    assertEquality("abcA(?i:[^a-bcd-e]{1,}a{1,2}b{1}(?i:cd*400+)20+)AC*", x6.toString());
+}
+
+function testInvalidInsertionsInRegExp() {
+    string a = "*";
+    string:RegExp|error x1 = trap re `${a}`;
```

Comment:
Is this the way we are usually testing runtime errors?

Reply:
this is similar to how we test runtime panics with expressions like additive expressions right? since we know this will cause a panic we can trap it?